### PR TITLE
feat(files): add possibility to redirect to slash-ended path

### LINKED ISSUE #671: New `index_file` handling doesn't distinguish between `directory` and `directory/`
The code on the 0.8 branch serves `directory/index.html` for requests to both `/directory` and `/directory/`. That doesn't work properly for `/directory` because it breaks relative links in `directory/index.html`. E.g. if `directory/index.html` contains `<a href="hello.html">`, that should resolve to `/directory/hello.html`, but when `directory/index.html` is served for the URL `/directory`, the link will resolve to `/hello.html`, which is wrong.

### LINKED ISSUE #1132: Slash needed when accessing a folder
Related to #661 and probably #241 

Given this service:
```rust
.service(
            Files::new("/", "./www")
                .index_file("index.html")
                .show_files_listing()
                .default_handler(web::to(|| {
                    NamedFile::open("./www/404.html")
                        .and_then(|x| Ok(x.set_status_code(StatusCode::NOT_FOUND)))
                })),
)
```

Whenever I browse to `localhost:3000/beta`, which has an `index.html` in it, I get the html but all the referenced `.css` files and images are retrieved as if the file was being served from the `root`, that is, `localhost:3000/`, and hence, not being retrieved at all.

 If I put a final slash in the path `localhost:3000/beta/` everything works ok.

I think maybe `actix-web` should insert a final slash whenever it detects that we're dealing with a folder so we could fix this issue.

### PR DESCRIPTION
**Pull request by robertohuertasm**
When accessing to a folder without a final slash, the index file will be loaded ok, but if it has
references (like a css or an image in an html file) these resources won't be loaded correctly if
they are using relative paths. In order to solve this, this PR adds the possibility to detect
folders without a final slash and make a 302 redirect to mitigate this issue. The behavior is off by
default. We're adding a new method called `redirect_to_slash_directory` which can be used to enable
this behavior.

Fixes #1132
Fixes #671

### CODE REVIEW

File: actix-files/CHANGES.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
 * Bump up `percent-encoding` crate version to 2.1
 
 * Allow user defined request guards for `Files` #1113
+  
+* Add option to redirect to a slash-ended path `Files` #1132
```
Comment (fafhrd91) [inline actix-files/CHANGES.md]: create new version entry, 0.1.5 is released already

**Comment by fafhrd91** [comment]
I think this Pr should solve #671 as well

**Review by fafhrd91 — APPROVED** [review]

File: actix-files/CHANGES.md (hunk 1 above)
Comment (robertohuertasm) [reply actix-files/CHANGES.md]: Which version would that be? 0.2.0 or 0.1.6?

Comment (fafhrd91) [reply actix-files/CHANGES.md]: 0.1.6

Comment (robertohuertasm) [reply actix-files/CHANGES.md]: Done! Didn't know the release date so I put a random date in the future.

Comment (JohnTitor) [reply actix-files/CHANGES.md]: It may be confused, you can put `UNRELEASED` or something else.

File: actix-files/CHANGES.md [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+
 ## [0.1.5] - 2019-10-08
 
 * Bump up `mime_guess` crate version to 2.0.1
 
 * Bump up `percent-encoding` crate version to 2.1
 
 * Allow user defined request guards for `Files` #1113
-
+
```
Comment (JohnTitor) [inline actix-files/CHANGES.md:14]: This line is unnecessary change.

File: actix-files/CHANGES.md (hunk 1 above)
Comment (robertohuertasm) [reply actix-files/CHANGES.md]: I changed it to TBD (to be defined)

File: actix-files/CHANGES.md (hunk 2 above)
Comment (robertohuertasm) [reply actix-files/CHANGES.md:14]: I removed spaces in all lines needed.

Comment (JohnTitor) [reply actix-files/CHANGES.md:14]: Hmm, this line is still unchanged.

Comment (robertohuertasm) [reply actix-files/CHANGES.md:14]: 🤦‍♂ my bad! solved

**Review by JohnTitor — APPROVED** [review]
Thanks!

**Comment by fafhrd91** [comment]
thanks!
